src/misc/util/utilFile.c

```diff
@@ -25,6 +25,23 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 
+// Handle legacy macros
+#if !defined(S_IREAD)
+#if defined(S_IRUSR)
+#define S_IREAD S_IRUSR
+#else
+#error S_IREAD is undefined
```

Comment:
Isn't it `S_IRUSR` which is undefined?

Reply:
No. `S_IREAD` and `S_IWRITE` are the obsolete macros, see: https://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html